Buddy Death

randomPoison opened the pull request:
- Pressing 'K' kills buddies.
- Once dead, buddies to not move and will not accept resources.

randomPoison committed:
Add Dying And Death Animation
- Updated buddy animations to have states for dying and death.

randomPoison committed:
Add Kill Option For Buddies
- Added BuddyStats.Kill() method for killing buddies. It plays the death animation and ends the behavior tree. Buddies do not accept resources when dead.

randomPoison committed:
Don't Give Dead Buddies Resources
- Also made the death animation a bit better. But only a bit.

brickman1444 commented on Assets/Scripts/Actors/BuddyStats.cs:
```diff
@@ -13,10 +19,21 @@ void Awake()
 	{
-		currentStats++;
-		Emote( heartMaterial );
+		if ( _isAlive )
```
Put a comment or maybe an console error message saying that we should revisit this and put an assert in here.

brickman1444 commented on Assets/Scripts/Actors/BuddyStats.cs:
```diff
@@ -25,4 +42,11 @@ public void Emote( Material emoteMaterial )
+	{
+		_isAlive = false;
+		Destroy( GetComponent<AIController>() );
+		GetComponentInChildren<Animator>().SetTrigger( "isDead" );
```
Maybe leave a comment explaining what happens after you do this stuff.

brickman1444 commented:
That's everything.

randomPoison committed:
Add DebugUtils
- Added debug utils for doing assertions.
- Added documentation for BuddyStats.Kill() to explain its side effects.

randomPoison committed:
Merge remote-tracking branch 'origin/master' into lildude_behaviors

Conflicts:
	Assets/Scripts/Actors/ActorComponents/PlayerActorResources.cs
